### PR TITLE
Change RmdNodeData.RmdNodeList to slice of strings

### PR DESCRIPTION
**Pull request by csookarr**

### CODE REVIEW

File: pkg/controller/rmdworkload/rmdworkload_controller.go [hunk 1]
```diff
@@ -178,8 +179,8 @@ func (r *ReconcileRmdWorkload) findObseleteWorkloads(request reconcile.Request)
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	obseleteWorkloads := make(map[string]string)
 
-	for nodeName, namespace := range r.rmdNodeData.RmdNodeList {
-		address, err := r.getPodAddress(nodeName, namespace)
+	for _, nodeName := range r.rmdNodeData.RmdNodeList {
+		address, err := r.getPodAddress(nodeName, defaultNamespace)
```
Comment (nolancon) [inline pkg/controller/rmdworkload/rmdworkload_controller.go]: `getPodAddress` function should only accept one parameter (nodeName) as we are now assuming it will always be default. The `defaultNamespace` const can be used inside the function instead.

File: pkg/controller/rmdworkload/rmdworkload_controller.go [hunk 2]
```diff
@@ -210,7 +211,7 @@ func (r *ReconcileRmdWorkload) findTargetedNodes(request reconcile.Request, rmdW
 	// Loop through nodes listed in RmdWorkload Spec, add/update workloads where necessary.
 	for _, nodeName := range rmdWorkload.Spec.Nodes {
 		// Get node service address
-		address, err := r.getPodAddress(nodeName, r.rmdNodeData.RmdNodeList[nodeName])
+		address, err := r.getPodAddress(nodeName, defaultNamespace)
```
Comment (nolancon) [inline pkg/controller/rmdworkload/rmdworkload_controller.go]: same as above.

File: pkg/controller/rmdworkload/rmdworkload_controller.go [hunk 3]
```diff
@@ -240,8 +241,8 @@ func (r *ReconcileRmdWorkload) findRemovedNodes(request reconcile.Request, rmdWo
 	removedNodes := make(map[string]string)
 	rmdWorkloadName := rmdWorkload.GetObjectMeta().GetName()
 
-	for nodeName, namespace := range r.rmdNodeData.RmdNodeList {
-		address, err := r.getPodAddress(nodeName, namespace)
+	for _, nodeName := range r.rmdNodeData.RmdNodeList {
+		address, err := r.getPodAddress(nodeName, defaultNamespace)
```
Comment (nolancon) [inline pkg/controller/rmdworkload/rmdworkload_controller.go]: same as above.

File: pkg/controller/rmdworkload/rmdworkload_controller.go [hunk 4]
```diff
@@ -264,7 +265,7 @@ func (r *ReconcileRmdWorkload) findRemovedNodes(request reconcile.Request, rmdWo
 				}
 			}
 			if !nodeExistsOnRmdWorkloadSpec {
-				address, err := r.getPodAddress(nodeName, namespace)
+				address, err := r.getPodAddress(nodeName, defaultNamespace)
```
Comment (nolancon) [inline pkg/controller/rmdworkload/rmdworkload_controller.go]: same as above.

File: pkg/state/state.go [hunk 5]
```diff
@@ -7,23 +7,39 @@ import (
...
+		RmdNodeList: []string{},
 	}
 }
 
 // UpdateRmdNodeData() adds new data to map or updates existing node data
-func (nd *RmdNodeData) UpdateRmdNodeData(nodeName string, namespaceName string) {
-	nd.RmdNodeList[nodeName] = namespaceName
+func (nd *RmdNodeData) UpdateRmdNodeData(nodeName string) {
+	nodeInList := false
+	if len(nd.RmdNodeList) != 0 {
```
Comment (nolancon) [inline pkg/state/state.go]: Don't think we need this check before entering the for loop.

File: pkg/state/state.go [hunk 6]
```diff
@@ -7,23 +7,39 @@ import (
...
+		for _, node := range nd.RmdNodeList {
+			if nodeName == node {
+				nodeInList = true
```
Comment (nolancon) [inline pkg/state/state.go]: There's a few ways to do this. Simplest is probably to just `return` here if we find the node name. 
Then outside the for loop execute the `append` regardless, it will only ever get to that point if the node doesn't exist. Removes the need for `nodeInList` variable

File: pkg/state/state_test.go [hunk 7]
```diff
@@ -9,40 +9,22 @@ func TestUpdateRmdNodeData(t *testing.T) {
...
-		namespaceName  string
 		nodeData       RmdNodeData
-		expectedStates map[string]string
+		expectedStates []string
 	}{
 		{
-			name:          "test case 1 - namespace updated for particular node",
-			nodeName:      "example-node-1",
-			namespaceName: "example-name-space",
+			name:     "test case 1 - data added to empty RmdNodeData struct",
```
Comment (nolancon) [inline pkg/state/state_test.go]: Another test case or two here? Case where RmdNodeData struct is not empty, node already exists in list etc.

**Review by nolancon — COMMENTED** [review]
Thanks. Some small tidy ups recommended.

**Review by nolancon — COMMENTED** [review]
LGTM. Thanks @csookarr
